### ISSUE TITLE
support for nested records

### ISSUE THREAD

**Issue by junmipan** [alanctgardner/gogen-avro#1]
Great to see this tool in the making.  Would you consider adding support for nested records?
For example, the following schema contains a nested record.
{
  "type": "record",
  "name": "OuterRecord",
  "namespace": "...",
  "doc": "Schema containing record type",
  "fields": [
    {
      "name": "nested-record",
      "type": {
        "type": "record",
        "name": "NestedRecord",
        "doc": "...",
        "fields": [
          {
            "name": "some-attribute",
            "type": "string"
          }
        ]
      }
    }
  ]
}

**Comment by alanctgardner** [comment]
I made some changes, nested record definitions should work now. Let me know if you have any more issues!

**Comment by junmipan** [comment]
Wow!  That is a lightning fast resolution.  Thanks, Alan. Great work!

A few more items for you to consider:
1. enum
2. non string type union members (primitive & complex types)
3. This could be a bit demanding.  Allow multiple schema (files) as input.  And allow latter to reference former as type.  Generate each in its own Go file.  You might have figured where this came from (the Apache avro-tools for Java :-)

Ex.
1.    {
        "namespace" : "...",
        "name"      : "Priority",
        "doc"       : "...",
        "type"      : "enum",
        "symbols"   : [ "HIGH", "NORMAL", "LOW" ]
    }

2.    {
        "namespace" : "...",
        "name": "tags",
        "doc"       : "...",
        "type": [
            "null",
            {
              "type": "map",
              "values": "string"
            }
        ]
    }

**Comment by alanctgardner** [comment]
1) I've added support for enumerations now, they should work.
2) All types can be members of unions right now - I need to add some integration tests for unions of non-primitive types, but everything *should* work.
3) This is a pretty big chunk of work, but it's on the roadmap (after deserializing). I made a new issue to cover this request.

I'm going to close this issue since the original problem is resolved - feel free to open new issues for other feature requests or bugs!
